### ISSUE TITLE
no-global-assign does not check for functions

### ISSUE THREAD

**Issue by retorquere** [eslint/eslint#11835]
**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 12.4.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
	"env": {
		"browser": false,
		"es6": true
	},
	"parserOptions": {
		"ecmaVersion": 7
	},
	"globals": {
		"Zotero": "readonly",
		"Z": "readonly",
		"ZU": "readonly",
		"DOMParser": "readonly",
		"XPathResult": "readonly",
		"XMLSerializer": "readonly",
		"atob": "readonly",
		"text": "readonly",
		"attr": "readonly"
	},
	"rules": {
          "no-global-assign": "error"
	}
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}
```

```bash
eslint Vice.js
```

**What did you expect to happen?**

I expected the function declaration to be reported as an error, as it overrides an existing global

**What actually happened? Please include the actual, raw output from ESLint.**

The function declaration is ignored.

**Are you willing to submit a pull request to fix this bug?**

Yes.

**Comment by mysticatea** [comment]
Thank you for your report.

This is intentional behavior because it's a redeclaration rather than an assignment.
Use `no-redeclare` rule.

```js
/*eslint no-redeclare: [error, {builtinGlobals: true}] */
/*globals attr */

function attr() {} // → 'attr' is already defined.
```
